packages/@ember/template-compilation/index.ts

```diff
@@ -2,7 +2,7 @@ import { DEBUG } from '@glimmer/env';
 
 export { compile as compileTemplate } from 'ember-template-compiler';
 
-export let precompileTemplate;
+export let precompileTemplate: () => never | undefined;
```

Comment:
`never | undefined` is just `undefined`, since `never` is a subtype of *all* types. In this position it's the inverse of writing `any | undefined`, which would unify to `any` since `any` is the *super*-type of all types.

```suggestion
export let precompileTemplate: () => undefined;
```

I *do* wish TS had a way to annotate that something would throw under certain conditions, but alas: no.

Reply:
This is wrong, it should be: `export let precompileTemplate: (() => never) | undefined`

Reply:
Ahhh, 👍🏼

Reply:
Well… actually… in that case you're saying that *no* actual invocation of `precompileTemplate` *ever* returns?

Reply:
Ah, so *locally* that is true, but it seems like in the *other* places this is used that'd be a bit weird? I don't actually understand what the intent of this is.